addon/authenticators/oauth2-password-grant.js

```diff
@@ -322,17 +322,24 @@ export default BaseAuthenticator.extend({
       merge(options.headers, clientIdHeader);
     }
     return new RSVP.Promise((resolve, reject) => {
-      fetch(url, options).then((response) => {
-        response.text().then((text) => {
-          let json = text ? JSON.parse(text) : {};
-          if (!response.ok) {
-            response.responseJSON = json;
-            reject(response);
-          } else {
-            resolve(json);
-          }
-        });
-      }).catch(reject);
+      fetch(url, options)
+        .then((response) => {
+          response.text().then((text) => {
+            let json;
+            try {
+              json = JSON.parse(text);
+            } catch (e) {
+              json = {};
+            }
+            if (response.status === 200) {
```

Comment:
Can we continue using `response.ok` here?

Reply:
If I'm right, response.ok does mean response.status >= 200 && response.status < 300 according to ![fetch](https://github.com/github/fetch), but RFC6749 allows only 200 for successful response.

Reply:
Yeah, relying on 200 would be a breaking change though and we'd need to include a deprecation if we see any other status codes. I'm all for doing that but should be a different PR.

Reply:
It depends on you, I think this solution is fixing current bug. All responses, which are not in accordance with RFC6749 will be rejected.
But if you want, ill give there back response.ok;)

Reply:
Yes, please use `response.ok` for now - happy to merge another PR that adds a deprecation for status codes other than 200!